packages/function-tree/src/index.js

```diff
@@ -191,6 +191,11 @@ export class FunctionTree extends EventEmitter {
     this.runTree.on = this.on.bind(this)
     this.runTree.once = this.once.bind(this)
     this.runTree.off = this.removeListener.bind(this)
+
+    // When not extended
+    if (!returnInstance) {
```

Comment:
But isn't previous factory does exactly same? It just expose only runTree without internal stuff, isn't it?
```js
export default (contextProviders) => {
  return new FunctionTree(contextProviders).runTree
}
```